src/components/ec-currency-input/ec-currency-input.spec.js

```diff
@@ -24,10 +46,94 @@ describe('EcCurrencyInput', () => {
   }
 
   it('should render properly', () => {
-    const wrapper = mountCurrencyInput({
-      currencies,
-      value: {},
-    });
+    const wrapper = mountCurrencyInput();
+
     expect(wrapper.element).toMatchSnapshot();
   });
+
+  describe(':props', () => {
+    it('should render the label when the label  is given', () => {
+      const wrapper = mountCurrencyInput({ label: 'Currency Input' });
+      expect(wrapper.findByDataTest('ec-currency-input__label-text').exists()).toBe(true);
+    });
+
+    it('should render the note when the note  is given', () => {
+      const wrapper = mountCurrencyInput({ note: 'max 80 chars' });
+      expect(wrapper.findByDataTest('ec-currency-input__note').exists()).toBe(true);
+    });
+
+    it('should render the error when the errorMessage is given', () => {
+      const wrapper = mountCurrencyInput({ errorMessage: 'Random message' });
+
+      expect(wrapper.find('.ec-currency-input__currencies .ec-input-field__input--has-error').exists()).toBe(true);
```

Comment:
not sure this is the best selector :thinking: 

it might be a smell that classnames could be improved, but happy to move along if anybody else disagrees

for reference, check how this is done in Rails https://guides.rubyonrails.org/v3.2/active_record_validations_callbacks.html#customizing-error-messages-css